cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -204,6 +206,144 @@ func AvailablePackageSummaryFromChart(chart *models.Chart) (*corev1.AvailablePac
+	log.Infof("+helm GetAvailablePackageDetail %s", contextMsg)
+
+	// Retrieve namespace, chartID, version from the request
+	namespace := request.AvailablePackageRef.Context.Namespace
+	chartID := request.AvailablePackageRef.Identifier
+	version := request.Version
+
+	// After requesting a specific namespace, we have to ensure the user can actually access to it
+	// If checking the global namespace, allow access always
+	hasAccess := namespace == s.globalPackagingNamespace
+	if !hasAccess {
+		var err error
+		// If checking another namespace, check if the user has access (ie, "get secrets in this ns")
+		hasAccess, err = s.hasAccessToNamespace(ctx, namespace)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Unable to check if the user has access to the namespace: %s", err)
+		}
+		if !hasAccess {
+			// If the user has not access, return a unauthenticated response, otherwise, continue
+			return nil, status.Errorf(codes.Unauthenticated, "The current user has no access to the namespace %q", namespace)
+		}
+	}
+
+	// Unescape URI-encoded characters, like '%2F' that becomes '/'
+	unescapedChartID, err := url.QueryUnescape(chartID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable to decode chart ID chart: %v", chartID)
+	}
+	// TODO(agamez): support ID with multiple slashes, eg: aaa/bbb/ccc
+	chartIDParts := strings.Split(unescapedChartID, "/")
+	if len(chartIDParts) != 2 {
+		return nil, status.Errorf(codes.InvalidArgument, "Incorrect request.AvailablePackageRef.Identifier, currently just 'foo/bar' patters are supported: %s", chartID)
+	}
+
+	// Since the version is optional, in case of an empty one, fall back to get all versions and get the first one
+	// Note that the chart version array should be ordered
+	var chart models.Chart
+	if version == "" {
+		log.Infof("Requesting chart '%s' (latest version) in ns '%s'", unescapedChartID, namespace)
+		chart, err = s.manager.GetChart(namespace, unescapedChartID)
+	} else {
+		log.Infof("Requesting chart '%s' (version %s) in ns '%s'", unescapedChartID, version, namespace)
+		chart, err = s.manager.GetChartVersion(namespace, unescapedChartID, version)
+	}
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable to retrieve chart: %v", err)
+	}
+
+	availablePackageDetail, err := AvailablePackageDetailFromChart(&chart)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable to parse chart to an availablePackageDetail: %v", err)
+	}
+
+	return &corev1.GetAvailablePackageDetailResponse{
+		AvailablePackageDetail: availablePackageDetail,
+	}, nil
+}
+
+func AvailablePackageDetailFromChart(chart *models.Chart) (*corev1.AvailablePackageDetail, error) {
+	pkg := &corev1.AvailablePackageDetail{}
+
+	if chart.Name == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Name not found on helm package: %v", chart)
+	}
+	pkg.Name = chart.Name
+	pkg.DisplayName = chart.Name
+
+	if chart.Icon == "" {
```

Comment:
Also listed as optional at https://helm.sh/docs/topics/charts/#the-chartyaml-file